packages/transactions/src/blockhash.ts

```diff
@@ -44,6 +44,26 @@ export function assertIsBlockhash(putativeBlockhash: string): asserts putativeBl
     }
 }
 
+function isBlockhashLifetimeTransaction(
+    transaction: BaseTransaction | (BaseTransaction & ITransactionWithBlockhashLifetime)
+): transaction is BaseTransaction & ITransactionWithBlockhashLifetime {
+    return (
+        'lifetimeConstraint' in transaction &&
+        typeof transaction.lifetimeConstraint.blockhash === 'string' &&
+        typeof transaction.lifetimeConstraint.lastValidBlockHeight === 'bigint'
+    );
+}
+
+export function assertIsBlockhashLifetimeTransaction(
+    transaction: BaseTransaction | (BaseTransaction & ITransactionWithBlockhashLifetime)
+): asserts transaction is BaseTransaction & ITransactionWithBlockhashLifetime {
+    if (!isBlockhashLifetimeTransaction(transaction)) {
+        // TODO: Coded error.
+        throw new Error('Transaction is not a blockhash transaction');
+    }
+    assertIsBlockhash(transaction.lifetimeConstraint.blockhash);
+}
```

Comment:
I'm not sure if it's on purpose but the logic of the `isX` and `assertIsX` functions is not the same. Passing a transaction with an invalid base58 blockhash will work for `isX` but fail for `assertIsIx`.

To make them the same, we could push that conditional to the `assertIsX` function and just do a try/catch on the `isX` function.

Reply:
Good point, thanks! I've refactored to move all the logic to the `isX` function, which makes them symmetrical as you say and also mirrors the durable nonce equivalent.